core/src/main/java/org/apache/calcite/util/NameMultimap.java

```diff
@@ -60,10 +82,21 @@ public NameMultimap() {
         && map.equals(((NameMultimap) obj).map);
   }
 
-  /** Adds an entry to this multimap. */
+  /**
+   * Adds an entry to this multimap. If allowsDuplicates is true, then the
+   * existing entry will be overwritten.
+   */
   public void put(String name, V v) {
     List<V> list = map().computeIfAbsent(name, k -> new ArrayList<>());
-    list.add(v);
+    int index = -1;
```

Comment:
Why do we need to perform all of this? Can't we just use a Set rather than an Array data structure if duplicates aren't allowed?

Reply:
I'll try changing it to a Set.